control/freqplot.py

```diff
@@ -1039,14 +1019,219 @@ def gangof4_plot(P, C, omega=None, **kwargs):
+    plot : bool
+        If True (default), generate the singular values plot.
+    omega_limits : array_like of two values
+        Limits of the frequency vector to generate.
+        If Hz=True the limits are in Hz otherwise in rad/s.
+    omega_num : int
+        Number of samples to plot.
+        Default value (1000) set by config.defaults['freqplot.number_of_samples'].
+    dB : bool
+        If True, plot result in dB.
+        Default value (False) set by config.defaults['singular_values_plot.dB'].
+    Hz : bool
+        If True, plot frequency in Hz (omega must be provided in rad/sec).
+        Default value (False) set by config.defaults['singular_values_plot.Hz']
+
+    Returns
+    -------
+    sigma : ndarray (or list of ndarray if len(syslist) > 1))
+        singular values
+    omega : ndarray (or list of ndarray if len(syslist) > 1))
+        frequency in rad/sec
+
+    Other Parameters
+    ----------------
+    grid : bool
+        If True, plot grid lines on gain and phase plots.  Default is set by
+        `config.defaults['singular_values_plot.grid']`.
+
+    Examples
+    --------
+    >>> den = [75, 1]
+    >>> sys = ct.tf([[[87.8], [-86.4]], [[108.2], [-109.6]]], [[den, den], [den, den]])
+    >>> omega = np.logspace(-4, 1, 1000)
+    >>> sigma, omega = singular_values_plot(sys)
+
+    """
+    # Make a copy of the kwargs dictionary since we will modify it
+    kwargs = dict(kwargs)
+
+    # Get values for params (and pop from list to allow keyword use in plot)
+    dB = config._get_param('singular_values_plot', 'dB', kwargs, singular_values_plot, pop=True)
+    Hz = config._get_param('singular_values_plot', 'Hz', kwargs, _bode_defaults, pop=True)
+    grid = config._get_param('singular_values_plot', 'grid', kwargs, _bode_defaults, pop=True)
+    plot = config._get_param('singular_values_plot', 'grid', plot, True)
+    omega_num = config._get_param('freqplot', 'number_of_samples', omega_num)
+
+    # If argument was a singleton, turn it into a tuple
+    if not hasattr(syslist, '__iter__'):
+        syslist = (syslist,)
+
+    omega, omega_range_given = _determine_frequency_range(syslist, omega, omega_limits, omega_num, Hz)
+    omega = np.atleast_1d(omega)
+
+    if plot:
+        fig = plt.gcf()
+        ax_sigma = None
+
+        # Get the current axes if they already exist
+        for ax in fig.axes:
+            if ax.get_label() == 'control-sigma':
+                ax_sigma = ax
+
+        # If no axes present, create them from scratch
+        if ax_sigma is None:
+            plt.clf()
+            ax_sigma = plt.subplot(111, label='control-sigma')
+
+        color_cycle = plt.rcParams['axes.prop_cycle'].by_key()['color']
+
+    sigmas, omegas, nyquistfrqs = [], [], []
+    for idx_sys, sys in enumerate(syslist):
+        omega_sys = np.asarray(omega)
+        if sys.isdtime(strict=True):
+            nyquistfrq = math.pi / sys.dt
+            if not omega_range_given:
```

Comment:
Issue a warning if a given omega is above nyquistfrq?

Reply:
Don't have a strong opinion about it. I added a warning for the singular value plot now (but then also the bode plot should have it I guess)

Reply:
Since the Nyquist frequency is plotted on the Bode/singular value plot, it seems like a warning may not be needed.  If you do add a warning, it would be nice to be able to turn it off (without having the user have to figure out how to filter warnings => keyword?).

Reply:
If everybody agrees, I would avoid issuing a warning at all to avoid additional logic that does not seem to add value.

Reply:
ok with  me to leave out warning